example/areaLight.js

```diff
@@ -0,0 +1,268 @@
+	samplesPerFrame: 1,
+	resolutionScale: 1 / window.devicePixelRatio,
+	filterGlossyFactor: 0.25,
+	tiles: 1,
+	multipleImportanceSampling: true
+
+};
+
+// clamp value for mobile
+const aspectRatio = window.innerWidth / window.innerHeight;
+if ( aspectRatio < 0.65 ) {
+
+	params.bounces = Math.min( params.bounces, 10 );
+	params.resolutionScale *= 0.5;
+	params.tiles = 3;
+
+}
+
+init();
+
+async function init() {
+
+	renderer = new THREE.WebGLRenderer( { antialias: true } );
+	renderer.toneMapping = THREE.ACESFilmicToneMapping;
+	document.body.appendChild( renderer.domElement );
+
+	camera = new THREE.PerspectiveCamera( 75, window.innerWidth / window.innerHeight, 0.025, 500 );
+	camera.position.set( 0.4, 0.6, 2.65 );
+
+	ptRenderer = new PathTracingRenderer( renderer );
+	ptRenderer.camera = camera;
+	ptRenderer.material = new PhysicalPathTracingMaterial();
+	ptRenderer.tiles.set( params.tiles, params.tiles );
+
+	fsQuad = new FullScreenQuad( new THREE.MeshBasicMaterial( {
+		map: ptRenderer.target.texture,
+	} ) );
+
+	controls = new OrbitControls( camera, renderer.domElement );
+	controls.target.set( - 0.15, 0.33, - 0.08 );
+	camera.lookAt( controls.target );
+	controls.addEventListener( 'change', () => {
+
+		ptRenderer.reset();
+
+	} );
+	controls.update();
+
+	camera.lookAt( - 0.15, 0.33, - 0.08 );
+
+	samplesEl = document.getElementById( 'samples' );
+
+	const envMapPromise = new Promise( resolve => {
+
+		new RGBELoader()
+			.load( 'https://raw.githubusercontent.com/mrdoob/three.js/dev/examples/textures/equirectangular/royal_esplanade_1k.hdr', texture => {
+
+				ptRenderer.material.envMapInfo.updateFrom( texture );
+				resolve();
+
+			} );
+
+	} );
+
+	const group = new THREE.Group();
+
+	const floorGeom = new THREE.PlaneBufferGeometry( 10, 10 );
+	const floorMat = new THREE.MeshPhysicalMaterial( { color: new THREE.Color( 0x999999 ), roughness: 0.1 } );
+	const floor = new THREE.Mesh( floorGeom, floorMat );
+	floor.rotateX( Math.PI / 2 );
+	group.add( floor );
+
+	const redBoxGeom = new THREE.BoxBufferGeometry( 0.5, 2, 2 );
+	const redBoxMat = new THREE.MeshPhysicalMaterial( { color: new THREE.Color( 0xAA0000 ) } );
+	const redBox = new THREE.Mesh( redBoxGeom, redBoxMat );
+	redBox.position.set( - 1.5, 1.0, 0.0 );
+	group.add( redBox );
+
+	group.updateMatrixWorld();
+
+	const areaLight1 = new THREE.RectAreaLight( new THREE.Color( 0xFFFFFF ), 50.0, 1.0, 1.0 );
+	areaLight1.position.x = 1.5;
+	areaLight1.position.y = 1.0;
+	areaLight1.position.z = - 0.5;
+	areaLight1.rotateZ( - Math.PI / 4 );
+	areaLight1.rotateX( - Math.PI / 2 );
+	group.add( areaLight1 );
+
+	const areaLight2 = new THREE.RectAreaLight( new THREE.Color( 0x00FF00 ), 20.0, 5.0, 1.0 );
+	areaLight2.position.x = - 2.5;
+	areaLight2.position.y = 0.5;
+	areaLight2.position.z = - 3.0;
+	areaLight2.rotateX( Math.PI );
+	group.add( areaLight2 );
+
+	areaLights = [ areaLight1, areaLight2 ];
+
+	transformControls = new TransformControls( camera, renderer.domElement );
```

Comment:
Can we add the ability to rotate and scale the light, as well? With the ability to change transform modes using W, E, and R for translate, rotate, and scale respectively?

Reply:
```js
window.addEventListener( 'keydown', function ( event ) {

  switch ( event.keyCode ) {
  
	  case 87: // W
		  transformControls.setMode( 'translate' );
		  break;
  
	  case 69: // E
		  transformControls.setMode( 'rotate' );
		  break;
  
	  case 82: // R
		  transformControls.setMode( 'scale' );
		  break;
  
  }

} ); 
```